ddp-train-ue.py

```diff
@@ -0,0 +1,372 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the license found in the
+# LICENSE file in the root directory of this source tree.
+
+"""Train a (really simple) universal facial encoder."""
+
+import argparse
+import copy
+import itertools
+import logging
+import os
+import platform
+import sys
+import time
+from collections import defaultdict
+from typing import Dict, Union
+
+import numpy as np
+import pandas as pd
+import torch
+import torch.distributed as dist
+import torch.multiprocessing as mp
+import torch.optim.lr_scheduler as lr_scheduler
+import torch.utils.data
+import yaml
+from fvcore.common.config import CfgNode as CN
+from PIL import Image
+from torch.nn.parallel import DistributedDataParallel as DDP
+from torch.utils.data.distributed import DistributedSampler
+from torch.utils.tensorboard import SummaryWriter
+
+from data.ava_dataset import none_collate_fn
+from data.headset_dataset import MultiCaptureDataset as HeadsetMultiCaptureDataset
+from models.headset_encoders.loss import UniversalEncoderLoss
+from models.headset_encoders.universal import UniversalEncoder
+from utils import load_checkpoint, tocuda, train_headset_csv_loader
+
+os.environ["TF_CPP_MIN_LOG_LEVEL"] = "1"
+sys.dont_write_bytecode = True
+
+
+class HostnameFilter(logging.Filter):
+    hostname = platform.node()
+
+    def filter(self, record):
+        record.hostname = HostnameFilter.hostname
+        return True
+
+
+root = logging.getLogger()
+root.setLevel(logging.INFO)
+handler = logging.StreamHandler(sys.stdout)
+handler.setLevel(logging.INFO)
+handler.addFilter(HostnameFilter())
+formatter = logging.Formatter("%(asctime)s - %(message)s")
+handler.setFormatter(formatter)
+root.addHandler(handler)
+
+torch.backends.cudnn.benchmark = True  # gotta go fast!
+
+
+def gen_optimizer(
+    net: torch.nn.Module,
+    optim_type: str,
+    batchsize: int,
```

Comment:
Can we refactor this with ddp-train.py?

Reply:
I tried but the name "ddp-train" cannot be imported due to the dash "-". Should be possible with importlib, if you think we should use that? (Or we can rename "ddp-train" to "ddp_train"...)

Reply:
Oh, I was thinking of extracting it to a common 3rd file